### PR TITLE
foot 1.18.1 (new formula)

### PR DESCRIPTION
**Pull request by david**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/f/foot.rb
Comment (chenrui333) [inline Formula/f/foot.rb:18]: need to add depends_on "freetype"

Comment (david) [reply Formula/f/foot.rb]: @chenrui333 done, thanks.

**Review by nandahkrishna — APPROVED** [review]
Thanks!

File: Formula/f/foot.rb
Comment (carlocab) [inline Formula/f/foot.rb:22]:
```suggestion
    ENV.O3
```
Is there a reason why we shouldn't be using Homebrew's default optimisation level here?

File: Formula/f/foot.rb
Comment (carlocab) [inline Formula/f/foot.rb:36]: Let's follow the style used by other formulae, please.
```suggestion
    system "meson", "setup", "build", "-Dterminfo-base-name=foot-extra", *std_meson_args
    system "meson", "compile", "-C", "build", "--verbose"
    system "meson", "install", "-C", "build"
```

File: Formula/f/foot.rb
Comment (carlocab) [inline Formula/f/foot.rb:52]:
```suggestion
      shell_output("#{bin}/foot --check-config 2>&1"),
```

File: Formula/f/foot.rb
Comment (carlocab) [inline Formula/f/foot.rb:60]:
```suggestion
    assert_empty shell_output("#{bin}/foot --check-config")
```

**Review by carlocab — COMMENTED** [review]
Thanks; a few comments.

File: Formula/f/foot.rb
Comment (david) [reply Formula/f/foot.rb:36]: Although this is much better (my apologies, I should have looked for other examples of meson use). 

Unfortunately, `foot` expects a couple of libraries created by the same author to be present, and when they are not, it falls back to cloning them as subprojects and using that. Since we're using `--wrap-mode=nofallback`, that doesn't happen. These libraries are not packaged in homebrew. 

Is it ok for me to remove the use of `--wrap-mode=nofallback` or do I need to package the other libraries?

File: Formula/f/foot.rb
Comment (david) [reply Formula/f/foot.rb:24]: It was recommended by the author for GCC 10, which doesn't apply anymore, so I removed it.

File: Formula/f/foot.rb
Comment (david) [reply Formula/f/foot.rb]: Changed, with a small tweak for the return value.

File: Formula/f/foot.rb
Comment (david) [reply Formula/f/foot.rb]: Done.

**Comment by david** [comment]
Note: I added harfbuzz and utf8proc as new dependencies.

File: Formula/f/foot.rb
Comment (carlocab) [reply Formula/f/foot.rb:38]: We can use `--wrap-mode=nodownload`, but we can't remove use of `--wrap-mode` entirely. (It's in `std_meson_args` for a reason, after all.) You'll have to either add the required dependencies as separate formulae or as `resource`s in this one.

Comment (david) [reply Formula/f/foot.rb:38]: Ok, I added #192886, and #192887, which will need to be merged first, and updated the dependencies for this formula.

**Review by chenrui333 — APPROVED** [review]

**Comment by david** [comment]
This depends on #192887  which is waiting for approval.

**Comment by david** [comment]
All tests passed now that https://github.com/Homebrew/homebrew-core/pull/192887 was merged, so moving this back to ready for review.

**Review by carlocab — APPROVED** [review]
Thanks!

**Pull request merged**
